### ISSUE TITLE
error when adding file to user library 

### ISSUE THREAD

**Issue by bernt-matthias** [galaxyproject/galaxy#5313]
I'm trying to add files from an ftp upload folder to a library. If the option `Convert line endings to POSIX` is selected an error appears. 

```
Traceback (most recent call last):
  File "/gpfs1/data/galaxy_server/galaxy-dev/tools/data_source/upload.py", line 415, in <module>
    __main__()
  File "/gpfs1/data/galaxy_server/galaxy-dev/tools/data_source/upload.py", line 402, in __main__
    add_file(dataset, registry, json_file, output_path)
  File "/gpfs1/data/galaxy_server/galaxy-dev/tools/data_source/upload.py", line 286, in add_file
    line_count, converted_path = sniff.convert_newlines(dataset.path, in_place=in_place, tmp_dir=tmpdir, tmp_prefix=tmp_prefix)
  File "/gpfs1/data/galaxy_server/galaxy-dev/lib/galaxy/datatypes/sniff.py", line 119, in convert_newlines
    fd, temp_name = tempfile.mkstemp(prefix=tmp_prefix, dir=tmp_dir)
  File "/global/apps/bioinf/galaxy/bin/Python-2.7.13/lib/python2.7/tempfile.py", line 314, in mkstemp
    return _mkstemp_inner(dir, prefix, suffix, flags)
  File "/global/apps/bioinf/galaxy/bin/Python-2.7.13/lib/python2.7/tempfile.py", line 244, in _mkstemp_inner
    fd = _os.open(file, flags, 0600)
OSError: [Errno 13] Permission denied: '/gpfs1/data/galaxy_server/galaxy-dev/database/files/004/data_id_4518_convert_rwcgHS'
```

`upload.py` is run on real user on our Galaxy. So I guess the problem is that the tempfile is created in `database/files`. 

One question: it is possible to select `Link files instead of copying` and `Convert line endings to POSIX` at the same time. What would happen if both are selected? Is the original file modified?

**Comment by nsoranzo** [comment]
@bernt-matthias Is this with run-as-real-user? Also, which Galaxy version? This might have been fixed already in 18.01.

**Comment by bernt-matthias** [comment]
Was with dev, but seems to work now. Can't reproduce it at the moment.
